docs/USAGE-GUIDE.md

```diff
@@ -1777,6 +1777,216 @@ handle. The handle is considered invalid after **s2n_connection_free** is used.
+
+**s2n_psk_set_secret** sets the secret value for a given PSK. Note that the secret value is copied into s2n-tls memory and the user is responsible for freeing the memory associated with the **secret** input.
+
+**s2n_psk_set_hmac** sets the PSK hash algorithm for a given PSK.
+The supported PSK hash algorithms are listed in the **s2n_psk_hmac** enum. This API overrides the default PSK hash algorithm value of **S2N_PSK_HMAC_SHA256**. Use this API if you wish to select a different cipher suite, as the selected cipher suite must have the same hash algorithm as the PSK.
+
+After adding an identity and secret to the PSK object, append the PSK to the connection using **s2n_connection_append_psk**. This API sets up the list of PSKs on the connection that will later be used by the PSK extension to select a PSK for the connection. If a PSK with a duplicate identity is found, an error is returned and the PSK is not added to the list. If the connection already contains a list of PSKs **s2n_connection_append_psk** will verify that the PSK mode being set matches that of the list. If the PSK modes differs, an error is returned. If there is no existing list of PSKs present on the connection (the input PSK is the first PSK to be added to the list) the PSK mode of the list is set to the input PSK's mode. The PSK object is copied to the connection and the user is responsible for freeing the memory associated with the PSK object. Note that both the server and client should order the PSKs appended in preference order ranging from highest preferred PSK to lowest preferred PSK.
+
+**s2n_config_set_psk_mode** sets the PSK mode on the config object. The PSK mode represents how the pre-shared key was generated and shared between the server and the client. The enum **s2n_psk_mode** lists the supported PSK modes. Use this API prior to the ClientHello to set the PSK mode for the connection. This API can also be used to switch between the different PSK modes.
+
+**s2n_connection_set_psk_mode** sets the PSK mode on the connection object. The supported PSK modes are listed in the
+enum **s2n_psk_mode** within the **api/s2n.h** file. This API overrides the PSK mode set on config for this connection.
+If the connection already contains a list of PSKs **s2n_connection_set_psk_mode** will verify that the PSK mode being set matches
+that of the list. If the PSK mode being set differs from the existing list, an error is returned. If there is no existing list of PSKs present on the connection, the PSK mode of the list is set to the input PSK mode.
+
+### Obtaining Negotiated Pre-Shared Keys
```

Comment:
Can you move this section to after the offered PSK section? It's weird that this comes first given that you should choose a PSK before you check with PSK was negotiated.

Reply:
It is not a part of the offered PSKs, this is the negotiated PSK selected by the server. It would not be correct to combine it with the offered PSK APIs.

Reply:
Move it _after_ the offered psk section.